test/TesterInternal/StorageTests/PersistenceProviderTests.cs

```diff
@@ -131,43 +131,67 @@ public void Dispose()
-            var cfg = new ProviderConfiguration(providerCfgProps, null);
-            await store.Init(testName, storageProviderManager, cfg);
+            var grainState = TestStoreGrainState.NewRandomState(stringLength).State;
 
-            await Test_PersistenceProvider_WriteRead(testName, store);
+            var store = await InitAzureTableStorageProvider(useJson, testName);
+
+            await Test_PersistenceProvider_WriteRead(testName, store, grainState);
         }
 
-        [Fact, TestCategory("Functional"), TestCategory("Persistence"), TestCategory("Azure")]
-        public async Task PersistenceProvider_Azure_WriteRead_Json()
+        [Theory, TestCategory("Functional"), TestCategory("Persistence"), TestCategory("Azure")]
+        [InlineData(null, "true", "false")]
+        [InlineData(null, "false", "true")]
+        [InlineData(15 * 32 * 1024 - 256, "true", "false")]
+        [InlineData(15 * 32 * 1024 - 256, "false", "true")]
+        public async Task PersistenceProvider_Azure_WriteRead_ChangeFormat(int? stringLength, string useJsonForWrite,
+            string useJsonForRead)
         {
-            string testName = "PersistenceProvider_Azure_WriteRead_Json";
+            var testName = string.Format("PersistenceProvider_Azure_WriteRead_ChangeFormat(stringLength = {0}, useJsonForWrite = {1}, useJsonForRead = {2})",
+                stringLength == null ? "default" : stringLength.ToString(), useJsonForWrite, useJsonForRead);
 
-            IStorageProvider store = new AzureTableStorage();
-            providerCfgProps.Add("DataConnectionString", StorageTestConstants.DataConnectionString);
-            providerCfgProps.Add("UseJsonFormat", "true");
-            var cfg = new ProviderConfiguration(providerCfgProps, null);
-            await store.Init(testName, storageProviderManager, cfg);
+            var grainState = TestStoreGrainState.NewRandomState(stringLength).State;
+            var grainId = GrainId.NewId();
 
-            await Test_PersistenceProvider_WriteRead(testName, store);
+            var store = await InitAzureTableStorageProvider(useJsonForWrite, testName);
+
+            var state = await Test_PersistenceProvider_WriteRead(testName, store,
+                grainState, grainId);
+
+            store = await InitAzureTableStorageProvider(useJsonForRead, testName);
+
+            await Test_PersistenceProvider_Read(testName, store, state, grainId);
         }
 
-        [Fact, TestCategory("Functional"), TestCategory("Persistence"), TestCategory("Azure")]
-        public void AzureTableStorage_ConvertToFromStorageFormat()
+        [Theory, TestCategory("Functional"), TestCategory("Persistence"), TestCategory("Azure")]
+        [InlineData(null, "false")]
+        [InlineData(null, "true")]
+        [InlineData(15 * 64 * 1024 - 256, "false")]
+        [InlineData(15 * 32 * 1024 - 256, "true")]
+        public async Task AzureTableStorage_ConvertToFromStorageFormat(int? stringLength, string useJson)
```

Comment:
I apologize for being unclear in my request when I requested a test that did:
>Swap between json and binary using large state
 ◦Write large (more than 192k) grain state as json string
 ◦Change to binary serialization.
 ◦Read state, verify state, Write state
 ◦Change to json serialization.
◦Read state, verify state

The reason I requested reading and writing both json and binary data into the same entity (as opposed to the conversion test here) is that I wanted to be sure that writing the new converted data would correctly clear out the old data.  That is if a entity had binary data in it, when it was saved as json, and reread, we should no longer see the binary data, but now see the json data.  This means we need to generate new data each write and write and read both types of data to the same entity.

The test submitted verifies that we can read the data regardless of what format it was written in, but does not verify that when writing data of a new format, the old data is correctly cleared out.